### DISCUSSION TITLE
Saving polymers as .mol, .xyz, and .pdb files

### DISCUSSION THREAD

**Discussion by mathersrob** [TheHydrophobicityProject/HydrophobicityExplorer#11] [Ideas]
@scohenjanes5,

I've noticed that .mol, .pdb, and .xyz files are not including hydrogens.

For example, the command line "makepol -n 5 -m Ethylene -p -v -c XMHP -s PE_5mer.xyz" outputs 5 xyz files (n=1-5).

Here is the output for PE_5mer_5.xyz

10
conformer_0
C     -2.527700    0.997700   -3.793300
C     -1.657100    0.449900   -2.672600
C     -2.488400   -0.325700   -1.649700
C     -1.642400   -0.993800   -0.561200
C     -0.924400    0.008100    0.346600
C     -0.244800   -0.648300    1.553200
C      0.837100   -1.681600    1.222700
C      1.986800   -1.179100    0.343400
C      2.755100    0.027900    0.886600
C      3.422600   -0.231500    2.228100

**Comment by scohenjanes5** [comment]
Thanks for bringing this up. This was not always the behavior, as I have examples in my filesystem with hydrogen atoms included. I found that there was an external command being used that removes Hs by default, and I'm not sure if that changed recently or not, but that's fixed now in the new branch I made. I have not yet moved those changes over to main because of other issues I noticed:

I also noticed that the calculations of properties was hanging in the example you sent. I'm looking into that now.

After I packaged the edits I did make, the packaged version failed to run because of something to do with the PIL module. I may have to look into a new way to package for conda, because this was not an issue when using setup.py to install locally.

**Comment by mathersrob** [comment]
Nice work Sander!

Btw, Seth mentioned the PIL error when he installed the package in December/January. At that time, I found a suggestion: https://github.com/rdkit/rdkit/issues/1179

Using pip install pillow worked for Seth. I had forgotten about that...

**Reply by scohenjanes5** [reply]
I ended up switching to showing the images with matplotlib, which allows me to remove the line importing PIL. 

I created a merge request with my edits and requested you to review it. I also attached a lengthy comment about the packaging process. The tldr of that is that it is trivially easy to package this project for pypi, but is onerous for conda.

I also had an issue with the WSL system I was initially using for development, which made it seem like the whole project was falling apart, but luckily it was just that system and not the package itself at fault.

**Reply by mathersrob** [reply]
Nice idea to use matplotlib. The comment about the packaging process is helpful.
